### ISSUE TITLE
Support correlating MULTISET operators in dialects that can't correlated subqueries from derived tables

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12045]
Regrettably, quite a few SQL dialects cannot correlate a subquery from within a derived table in that subquery:

```sql
SELECT (
  SELECT u.x
  FROM (
    SELECT t.x -- No clue what this is
  ) u
)
FROM t
```

These dialects include:

- [x] Db2
- [x] H2
- [x] MariaDB (see https://jira.mariadb.org/browse/MDEV-28196)
- [x] MySQL 5.7 (8 can handle it)
- [x] Oracle 11g (12c can handle it)

Others are also affected, but don't support SQL/JSON or SQL/XML, so we don't emulate `MULTISET` on those, yet. We need a better emulation that avoids the derived table if possible, for those dialects. Without a derived table, some other features will be much harder to support from within `MULTISET(...)`, including:

- Set operations (`UNION`, `EXCEPT`, `INTERSECT`)
- `DISTINCT`
- `ORDER BY`
- `OFFSET .. FETCH`

I.e. all the operations that happen after `SELECT` in the logical order of operations. Also, any of these operations are forbidden for the simplified emulation:

- `GROUP BY` / `HAVING` / aggregate functions
- `WINDOW` / `QUALIFY` / window functions

In other words, only these clauses are allowed:

- `SELECT`
- `FROM` (including all join types)
- `WHERE`

### Tasks

- [x] Support alternative emulations for these dialects
- [x] Update `DSL.multiset()` Javadoc, amending the limitation remark

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12129)

**Comment by lukaseder** [comment]
This emulation could be done using the new query object model transformation capabilities.

**Comment by lukaseder** [comment]
Looking into this now, for jOOQ 3.18. The simplified emulation does seem to work, this is exciting!

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.18. Further improvements will include support for `SELECT DISTINCT` in such simplified emulations. For this, we'll need: https://github.com/jOOQ/jOOQ/issues/10730.

The other limitations (no set operation support, no `GROUP BY` support, no aggregate/window function support, etc.) will remain.

**Comment by lukaseder** [comment]
https://github.com/jOOQ/jOOQ/issues/10730 has been implemented, and now, the `MULTISET(SELECT DISTINCT ..)` correlated subquery style can be emulated as well, if using the `JSON` emulation.
